qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -572,12 +573,15 @@ def update_toolbar_waterfall_plot(self, is_waterfall):
         self.toolbar.set_generate_plot_script_enabled(not is_waterfall)
 
     def change_line_collection_colour(self, colour):
```

Comment:
I was thinking about using just the method `.findobj()` from matplotlib to rewrite this function and this is what I came up with:
![image](https://github.com/mantidproject/mantid/assets/80104863/ea8dba17-0112-4b4a-a07d-aba90e8eb001)

This way I think the functions acts more generally and the methods `set_color()` and `set_edgecolor()` are always called on the correct collection. What do you think, is there something I might be missing?

Reply:
I just realised that I could just copy the text instead of a screenshot ...

Here it is:
```
    def change_line_collection_colour(self, colour):
        path_cols = self.canvas.figure.findobj(PathCollection)
        for path in path_cols:
            path.set_edgecolor(colour.name())

        line_cols = self.canvas.figure.findobj(LineCollection)
        for line in line_cols:
            line.set_color(colour.name())

        self.canvas.draw()
```

Reply:
Thanks for the suggestion. I agree is a more simple solution, there's a caveat. Initially I wanted to do something like this, but I though calling path or line collections blindly may not be much future proof. (For example the colorbar has line collections too, so you would retrieve it in calling it blindly with findobj). Right now is now a problem because wireframe does not have colorbar, but it may have some sort of unknown interactions in the future, that's why I tried to be more explicit. Anyway, I like this too, so I can do it like this, what do you think?

Reply:
I have changed it, I think it will be probably fine anyway. In case we change the function or we admit other plots to have line changes, this will be easy to detect if there is an issue. And your suggestions works better!

Reply:
Ah I see, that is a good point! What if we include the check to see if it's a wireframe or contour, like this:

```
    def change_line_collection_colour(self, colour):
        
        if figure_type(self.canvas.figure) == FigureType.Contour:
            path_cols = self.canvas.figure.findobj(PathCollection)
            for path in path_cols:
                path.set_edgecolor(colour.name())

        if figure_type(self.canvas.figure) == FigureType.Wireframe:
            line_cols = self.canvas.figure.findobj(LineCollection)
            for line in line_cols:
                line.set_color(colour.name())

        self.canvas.draw()
```

Do you think this is future proof?

Reply:
Thanks. It's updated